src/Learning/KWUserInterface/KWClassManagementActionView.cpp

```diff
@@ -300,6 +311,61 @@ void KWClassManagementActionView::SetObject(Object* object)
 	UIObjectView::SetObject(object);
 }
 
+void KWClassManagementActionView::RefreshHelpLists()
+{
+	UIList* classNameHelpList;
+	KWClass* kwcClass;
+	StringVector svClassNames;
+	StringVector svNonStorableClassNames;
+	int i;
+
+	assert(objValue != NULL);
+
+	// Liste des dictionnaires charges en memoire, par defaut non visible
+	classNameHelpList = cast(UIList*, GetFieldAt("ClassNames"));
+
+	// On commence vider la liste
+	classNameHelpList->RemoveAllItems();
+
+	// Collecte de classe stockables ou non a partir du domaine courant,
+	for (i = 0; i < KWClassDomain::GetCurrentDomain()->GetClassNumber(); i++)
+	{
+		kwcClass = KWClassDomain::GetCurrentDomain()->GetClassAt(i);
+		if (kwcClass->IsKeyBasedStorable())
+			svClassNames.Add(kwcClass->GetName());
+		else
+			svNonStorableClassNames.Add(kwcClass->GetName());
+	}
+
+	// Parametrage des classes stockables en premier, avec ligne blanche prealable
+	// comme pour le style "EditableComboBox" utilise dans les boites de dialogues
+	// ou la liste des dictionnaire est connue a l'avance
+	if (svClassNames.GetSize() > 0)
+	{
+		classNameHelpList->AddItem();
+		for (i = 0; i < svClassNames.GetSize(); i++)
+		{
+			// Ajout du nom de la classe dans la liste d'aide a la saisie
+			classNameHelpList->AddItem();
+			classNameHelpList->SetStringValueAt("Name", svClassNames.GetAt(i));
+		}
+	}
+
+	// Parametrage des classes non stockables ensuite, avec ligne blanche de separation
+	if (svNonStorableClassNames.GetSize() > 0)
+	{
+		assert(svClassNames.GetSize() > 0);
+		assert(classNameHelpList->GetItemNumber() >= 2);
```

Comment:
Why is this assert needed?

Reply:
Ajout d'un commentaire:
		// Il doit y avoir au moins une classe stockable, et donc au moins deux items dans
		// la liste en cours, ce qui fait qu'on n'aura jamais deux items vides de suite

Reply:
Ok, the new comment clarifies the assert IMHO. Hence, I can close this discussion thread.